pandas/_libs/tslibs/parsing.pyx

```diff
@@ -825,6 +825,10 @@ def format_is_iso(f: str) -> bint:
     iso_template = '%Y{date_sep}%m{date_sep}%d{time_sep}%H:%M:%S.%f'.format
     excluded_formats = ['%Y%m%d', '%Y%m', '%Y']
 
+    if (f is not None) and (f[-3:] in ["S%z", "S%Z", "f%z", "f%Z"]):
```

Comment:
Instead could we just loop over the possible timezone directives in the section below?
e.g.

```
iso_template = '%Y{date_sep}%m{date_sep}%d{time_sep}%H:%M:%S.%f{tz_dir}'.format
...
   for time_sep in ...:
       for tz_dir in ["", "%z", "%Z"]:
            ...
```

Reply:
Thank you for your comment.
Just to confirm, a loop version will be the following codes.
(I checked this passes the test)

```py
    iso_template = '%Y{date_sep}%m{date_sep}%d{time_sep}%H:%M:%S{micro_or_tz}'.format
    excluded_formats = ['%Y%m%d', '%Y%m', '%Y']

    for date_sep in [' ', '/', '\\', '-', '.', '']:
        for time_sep in [' ', 'T']:
            for micro_or_tz in ['', '%z', '%Z', '.%f', '.%f%z', '.%f%Z']:
                if (iso_template(date_sep=date_sep,
                                 time_sep=time_sep,
                                 micro_or_tz=micro_or_tz,
                                ).startswith(f) and f not in excluded_formats):
```

My concern was that this increase a nest a little.
But this version might be better for simplicity.

Reply:
Yes that's the loop I'm talking about. I think this version is more explicit rather than checking for the presence of `%z/Z` and then removing